compiler-rt/lib/tysan/tysan.cpp

```diff
@@ -128,6 +128,10 @@ static bool isAliasingLegalUp(tysan_type_descriptor *TDA,
           break;
       }
 
+      //You can't have negative offset, you must be partially inside the last type
```

Comment:
Space at start of comment; better to write the comment in a passive tone, "The offset can't be negative, thus we must be..."